src/modules/consumer.js

```diff
@@ -27,22 +32,53 @@ class Consumer {
    * @return {any}       object, string, number... the current received message
    */
   async checkRpc(messageProperties, queue, reply) {
-    if (messageProperties.replyTo) {
-      const options = {
-        correlationId: messageProperties.correlationId,
-        persistent: true,
-        durable: true,
-      };
-      this._connection.config.logger.debug({
-        message: `${loggerAlias} [${queue}][${messageProperties.replyTo}] > ${reply}`,
-        params: { content: reply },
-      });
+    if (!messageProperties.replyTo) {
+      return messageProperties;
+    }
+
+    const options = {
+      correlationId: messageProperties.correlationId,
+      persistent: true,
+      durable: true,
+    };
+    const serializedReply = parsers.out(reply, options);
+    await this.hooks.trigger(this, ConsumerHooks.beforeRpcReplyEvent, {
+      receiveProperties: messageProperties,
+      replyProperties: options,
+      queue,
+      reply,
+      serializedReply,
+      error: reply && reply.error ? reply.error : undefined,
+    });
+    this._connection.config.logger.debug({
+      message: `${loggerAlias} [${queue}][${messageProperties.replyTo}] > ${reply}`,
+      params: { content: reply },
+    });
 
+    try {
       const defaultChannel = await this._connection.getDefaultChannel();
-      return await defaultChannel.sendToQueue(messageProperties.replyTo, parsers.out(reply, options), options);
-    }
+      const written = await defaultChannel.sendToQueue(messageProperties.replyTo, serializedReply, options);
```

Comment:
according to amqplib types, this method just returns boolean and not `Promise<boolean>`.

Reply:
FYI from what I understand form the amqplib code this should always return true